.changeset/famous-ducks-smash.md

```diff
@@ -0,0 +1,5 @@
+---
+'@urql/exchange-graphcache': patch
+---
+
+Changes the type for the `updater` function of `cache.updateQuery` to `DataFields` this doesn't require `__typename` to be defined.
```

Comment:
```suggestion
Update the `updater` function type of `cache.updateQuery` to have a return type of `DataFields` so that `__typename` does not need to be defined.
```